ntoskrnl/mm/ARM3/virtual.c

```diff
@@ -4677,7 +4677,7 @@ NtAllocateVirtualMemory(IN HANDLE ProcessHandle,
         /* Fail without it */
         DPRINT1("Privilege not held for MEM_LARGE_PAGES\n");
         Status = STATUS_PRIVILEGE_NOT_HELD;
-        goto FailPathNoLock;
+        goto FailPathNoLockNoQuota;
```

Comment:
I would prefer to not add any more fail paths. IMO there should preferably be only one exit path, otherwise stuff gets too messed up. Use a local variable to track, whether you need too return quota instead.